### ISSUE TITLE
compiles and uploads, won't run. 

### ISSUE THREAD

**Issue by Racer1Tn** [lmarty/Hackerboxes20#13]
Program compiles and runs, SD card library copied too. No errors.
Will not boot, just a white screen. All other sketches run successfully.

**Comment by lmarty** [comment]
Hold down the flash button at boot time and see if that make a difference for you.  If so, you can solder a bridge across the j1 pads on the back of the tft and you wont have to do that anymore.

**Comment by Racer1Tn** [comment]
Holding button does not work for mine. :(

**Comment by lmarty** [comment]
Is your SD formatted using sd formatter?  https://www.sdcard.org/downloads/formatter_4/
Anything showing on the serial monitor?

**Comment by lmarty** [comment]
So here's a weird one.  How big is your SD card?  Testing on my 4g SD, things work fine.  Using a 32G, I see the same behavior you get here.  Blank white screen.  But I see the wifi scans taking place on the serial monitor after a bit.  At this point, I've got no idea why the size of the card would cause the screen to puke...

**Comment by cipioh** [comment]
So I have the same issue. Just a blank white display. I loaded the SD card libraries since they were there, but not using an SD card. Is this required?

**Comment by lmarty** [comment]
My badge will not boot without a SD with this code on it.  I have also not had any success with any card over 4GB.  Tested Numerous brands and cards, that seems to be the only common theme amongst them.  Hope that helps!

**Comment by cipioh** [comment]
Yeah... I looked at the ino when I had a little time, and realized the SD card was critical. I was hoping to see more of these with custom loads at DefCon...I did my own thing just playing around, but it looked like everyone else was just running the Badge B load.  It would have been really cool to have a more mature game sketch. Oh well...

**Comment by lmarty** [comment]
Yeah, me too man. I didn't have enough time beforehand to do anything exciting.  For such a capable little platform, I expected a lot more excitement and honestly, PRs. Since its so easy to code for.

**Comment by newgnu** [comment]
I too am guilty of running the Badge B code, but only 'cause I couldn't get any of my cards to work with this one.

Does yours work with any of the smaller cards? I've tried a handfull of different 1G/2G cards, and even an older 128M type 1 card. The only success I've had was with a 4G card (that I didn't find until _after_ DC), and it was still really temperamental even with the reset trick/jumper mod. I think it worked 3~4 times out of the dozen or so I tried it.

**Comment by newgnu** [comment]
Turns out my timing on the flash button was my problem, and the 'j1' pads on the back of the screen didn't matter if they were bridged or not.

It doesn't look like it can access the card at all unless that button is down either (at least on my badge). I threw together a POC that just loops around the SD.begin() and a message on the tft screen. After it initializes, it calls the same directory list code, but if I let go after the 'OK', I get a blank screen. if I continue to hold the button down, I get the list.

you can poke at https://github.com/newgnu/hb20_sd_test if you want to try yourself. I've even had success with my 32G card.
